sphinx_gallery/gen_rst.py

```diff
@@ -521,13 +541,13 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
     src_file = os.path.normpath(os.path.join(src_dir, fname))
     example_file = os.path.join(target_dir, fname)
     shutil.copyfile(src_file, example_file)
-    script_blocks = split_code_and_text_blocks(src_file)
+    file_conf, script_blocks = split_code_and_text_blocks(src_file)
     amount_of_code = sum([len(bcontent)
-                          for blabel, bcontent in script_blocks
+                          for blabel, bcontent, lineno in script_blocks
                           if blabel == 'code'])
 
     if md5sum_is_current(example_file):
-        return amount_of_code, 0
+        return get_docstring_and_rest(example_file)[0], amount_of_code, 0
```

Comment:
return script_blocks[0][1], like in the end of the function.
Even better we can return the introduction instead of the docstring, here and at the end. Then we can use the introduction directly in `generate_dir_rst`